### PR TITLE
validate JIRA issue key | SRENEW-1081

### PR DESCRIPTION
**Pull request by HarryGogonis**
- update package-lock version
- update packages
- fixes for updated packages
- implement jira check
- update CI
- now uses node 16

### CODE REVIEW

File: src/pr.ts [hunk 1]
```diff
@@ -1,33 +1,71 @@
...
 import {PullRequestEvent} from '@octokit/webhooks-types'
-import {Options} from './options'
+import {JiraClientImpl} from './jira'
+import {getInput, Options} from './options'
+
+export async function process(
+  context: Context,
+  isValid = validate
+): Promise<void> {
+  console.log(context.eventName)
```
Comment (HarryGogonis) [inline src/pr.ts:11]: remove console.log

File: package.json [hunk 2]
```diff
@@ -3,15 +3,15 @@
   "version": "0.0.0",
   "private": true,
   "description": "TypeScript template action",
-  "main": "lib/main.js",
+  "main": "src/main.ts",
```
Comment (guitarislife87) [inline package.json:6]: Should this be dist/index.js since that's what gets executed?

Comment (HarryGogonis) [reply package.json:6]: What gets executed is defined in `action.yml`

```yaml
runs:
  using: 'node16'
  main: 'dist/index.js'
```

Comment (HarryGogonis) [reply package.json:6]: src/main.ts is what `ncc` uses to bundle the dist folder

File: src/jira.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          email,
+          apiToken
+        }
+      }
+    })
+  }
+
+  async issueExists(issueIdOrKey: string): Promise<boolean> {
+    try {
+      await this.client.issues.getIssue({issueIdOrKey})
```
Comment (guitarislife87) [inline src/jira.ts:30]: Does this err if there's no issue with that id?

Comment (HarryGogonis) [reply src/jira.ts:30]: it 404s and throws if the issue does not exist

Comment (HarryGogonis) [reply src/jira.ts:30]: I suppose if it 401 auth error the issue could still exist, but not sure if that matters here

**Review by guitarislife87 — COMMENTED** [review]
Had 2 questions

File: src/jira.ts (hunk 3 above)
Comment (guitarislife87) [reply src/jira.ts:30]: Wow....horrible API design causing exception handling as flow control :smdh:

**Review by guitarislife87 — APPROVED** [review]
:lgtm:

**Pull request merged**